src/main/java/net/minecraftforge/client/event/RegisterPresetEditorsEvent.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.client.event;
+
+import java.util.Map;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.client.gui.screens.worldselection.PresetEditor;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.world.level.levelgen.presets.WorldPreset;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.LogicalSide;
+import net.minecraftforge.fml.event.IModBusEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+
+/**
+ * <p>Event for registering {@link PresetEditor} screen factories for world presets.</p>
+ *
+ * <p>This event is not {@linkplain Cancelable cancellable}, and does not {@linkplain HasResult have a result}.
+ *
+ * <p>This event is fired on the {@linkplain FMLJavaModLoadingContext#getModEventBus() mod-specific event bus},
+ * only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
+ */
+public class RegisterPresetEditorsEvent extends Event implements IModBusEvent
+{
+    private final Map<ResourceKey<WorldPreset>, PresetEditor> editors;
+    
+    @ApiStatus.Internal
+    public RegisterPresetEditorsEvent(Map<ResourceKey<WorldPreset>, PresetEditor> editors)
+    {
+        this.editors = editors;
+    }
+    
+    /**
+     * Registers a PresetEditor for a given workd preset key.
+     */
+    public void register(ResourceKey<WorldPreset> key, PresetEditor editor)
+    {
+        this.editors.put(key, editor);
```

Comment:
I don't know enough about usage people would have, but is this something that would make sense to validate against overriding? The main reason I ask is that because vanilla's entries get added before the event is fired which means that modders can override the preset editor's for vanilla presets which just feels *wrong* when the name itself specifies they are meant as a specific "preset"

Reply:
"PresetEditor" means editor of a WorldPreset, not editor-that-has-already-been-set.

I like the idea of allowing mods to override the vanilla editors here but we can open that to further discussion

Reply:
I will defer then to both you and other people who do world gen mods and the like what use cases there are and if it makes more sense for this to be overridable or not, but will give it at least a few days to give other devs a chance to weigh in on this